pennylane/devices/default_tensor.py

```diff
@@ -147,6 +147,10 @@ class DefaultTensor(Device):
     """A PennyLane device to perform tensor network operations on a quantum circuit using
     `quimb <https://github.com/jcmgray/quimb/>`_.
 
+    This device is designed to simulate large-scale quantum circuits using tensor networks. For small circuits, other devices like ``default.qubit`` are more suitable.
+    Different methods can be used to simulate the quantum circuit, and the backend uses the ``quimb`` library to perform the tensor network operations.
+    Currently, only the Matrix Product State (MPS) method is supported, based on the ``quimb``'s ``CircuitMPS`` class.
+
     Args:
         wires (int, Iterable[Number, str]): Number of wires present on the device, or iterable that
```

Comment:
Is `wires` a required argument?

Reply:
Thank you for the question. I would have raised this discussion next Monday (as this is a change of requirement from the first Epic), but I'm happy to talk about this now. Making `wires` a required argument was a change I made yesterday as a follow-up discussion with @albi3ro . 

In a nutshell, with `default.tensor` we need to create an initial state using the `MPS_computational_state` function in `quimb`, which requires the number of qubits to be defined (as well as the data type to use for the tensor). Therefore, if we allow the `wires` argument to be `None`, we have a very cryptic error from `quimb` as soon as we compute something with this device. Such an error is also raised if there's a mismatch between the circuit's wires and those defined on the device (so I had to intercept this with a check in the `simulate` function).

After some thought, I made the `wires` argument mandatory to avoid possible confusion, but if this is a problem we can revise this at the cost of adding some more checks

Reply:
Thanks @PietropaoloFrisoni! So the wire number can't be inferred at circuit runtime?

Reply:
Currently no, since we create an initial state with `quimb` before running the circuit and calling the `map_to_standard_wires` function. Therefore, right now the `wires` argument is the last positional argument (as in other devices like `lightning.qubit`)

Reply:
Thanks! Ok, not a blocker for this PR :thinking: Maybe I need to understand more, but it's not clear to me why the `quimb` state creation must happen at device instantiation and not at runtime. Was this a conscious design decision / constraint, or just something that's been overlooked?